src/System.Collections/src/System/Collections/Generic/Queue.cs

```diff
@@ -0,0 +1,461 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+/*=============================================================================
+**
+** Class: Queue
+**
+** Purpose: A circular-array implementation of a generic queue.
+**
+** Date: January 28, 2003
+**
+=============================================================================*/
+
+using System;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+
+namespace System.Collections.Generic
+{
+    // A simple Queue of generic objects.  Internally it is implemented as a 
+    // circular buffer, so Enqueue can be O(n).  Dequeue is O(1).
+    [DebuggerTypeProxy(typeof(QueueDebugView<>))]
+    [DebuggerDisplay("Count = {Count}")]
+    [System.Runtime.InteropServices.ComVisible(false)]
+    public class Queue<T> : IEnumerable<T>,
+        System.Collections.ICollection,
+        IReadOnlyCollection<T>
+    {
+        private T[] _array;
+        private int _head;       // First valid element in the queue
+        private int _tail;       // Last valid element in the queue
+        private int _size;       // Number of elements.
+        private int _version;
+        private Object _syncRoot;
+
+        private const int _MinimumGrow = 4;
+        private const int _ShrinkThreshold = 32;
+        private const int _GrowFactor = 200;  // double each time
+        private const int _DefaultCapacity = 4;
+        static T[] _emptyArray = new T[0];
```

Comment:
Thanks Stephen.  Code formatter fixed all the statics, and I manually fixed all the constants.